zipline/errors.py

```diff
@@ -191,6 +191,13 @@ class IncompatibleHistoryFrequency(ZiplineError):
 """.strip()
 
 
+class HistoryInInitialize(ZiplineError):
+    """
+    Raised when an algorithm calls history() in initialize.
+    """
+    msg = "history() can only be called in handle_data()"
```

Comment:
`before_trading_start()` behaves as expected in `history()`, *if* it is not called on the first day (in daily mode) or minute (in minutely mode), as in this case the `data` object is not populated and the algo will fail with an `IndexError`. Since we don't output a more informative message it probably can't be called **supported**. We should probably change `can` -> `should`, but not a big deal.